language/stdlib/modules/libra_coin.move

```diff
@@ -29,43 +33,40 @@ module LibraCoin {
         // this is to prevent the market cap's total value from hitting u64_max due to excessive
         // minting. This will not be a problem in the production Libra system because coins will
         // be backed with real-world assets, and thus minting will be correspondingly rarer.
-        // * 1000000 because the unit is microlibra
-        Transaction::assert(value <= 1000000000 * 1000000, 11);
-
+        Transaction::assert(value <= 1000000000 * 1000000, 11); // * 1000000 because the unit is microlibra
         // update market cap resource to reflect minting
         let market_cap = borrow_global_mut<MarketCap>(0xA550C18);
-        market_cap.total_value = market_cap.total_value + value;
+        market_cap.total_value = market_cap.total_value + (value as u128);
 
-        T { value }
+        T{ value: value }
     }
 
     // This can only be invoked by the Association address, and only a single time.
     // Currently, it is invoked in the genesis transaction
     public fun initialize() {
         // Only callable by the Association address
         Transaction::assert(Transaction::sender() == 0xA550C18, 1);
-
-        move_to_sender(MintCapability{});
-        move_to_sender(MarketCap { total_value: 0 });
+        move_to_sender<MintCapability>(MintCapability{ });
+        move_to_sender<MarketCap>(MarketCap { total_value: 0u128 });
     }
 
     // Return the total value of all Libra in the system
-    public fun market_cap(): u64 acquires MarketCap {
+    public fun market_cap(): u128 acquires MarketCap {
         borrow_global<MarketCap>(0xA550C18).total_value
     }
 
     // Create a new LibraCoin::T with a value of 0
     public fun zero(): T {
-        T { value: 0 }
+        T{value: 0}
     }
 
     // Public accessor for the value of a coin
-    public fun value(coin: &T): u64 {
-        coin.value
+    public fun value(coin_ref: &T): u64 {
```

Comment:
should we have and insist on the `_ref` part in our code base? I just ask because we don't do it in, say, our rust code base and in general when coding in almost any language, but we push it here and there in our move code.
I am not too sure but it feels a bit verbose and diminishes readability in my mind.
This is more a question for everybody and for the future rather than a request for change

Reply:
I personally am not a fan of this naming convention. In dynamic languages I can get behind them. But, in such a strongly-typed setting I don't think these names offer anything, and can in fact lead to issues (if e.g. the type is changed to no longer be a reference).

TL;DR: naming conventions (unlike the types) are not checked, we should rely on the checked stuff instead.